cores/arduino/pins_arduino.h

```diff
@@ -239,10 +292,14 @@ uint32_t pinNametoDigitalPin(PinName p);
 // Convert an analog pin number to a digital pin number
 #if defined(NUM_ANALOG_INPUTS) && (NUM_ANALOG_INPUTS>0)
 // Used by analogRead api to have A0 == 0
+#ifndef analogInputToDigitalPin
 #define analogInputToDigitalPin(p)  (((uint32_t)p < NUM_ANALOG_INPUTS) ? (p+A0) : p)
+#endif
 #else
+#ifndef analogInputToDigitalPin
```

Comment:
This have to be removed as this never happen.
If we are here then there is no Analog pin.
```suggestion
```